acceptance/localcluster/docker.go

```diff
@@ -190,32 +194,43 @@ func (c *Container) Start(binds []string, dns, vols *Container) error {
 	if vols != nil {
 		config.VolumesFrom = append(config.VolumesFrom, vols.ID)
 	}
-	return c.client.StartContainer(c.ID, config)
+	return c.cluster.client.StartContainer(c.ID, config)
 }
 
 // Pause pauses a running container.
 func (c *Container) Pause() error {
-	return c.client.PauseContainer(c.ID)
+	return c.cluster.client.PauseContainer(c.ID)
 }
 
 // Unpause resumes a paused container.
 func (c *Container) Unpause() error {
-	return c.client.UnpauseContainer(c.ID)
+	return c.cluster.client.UnpauseContainer(c.ID)
 }
 
 // Restart restarts a running container.
 // Container will be killed after 'timeout' seconds if it fails to stop.
 func (c *Container) Restart(timeoutSeconds int) error {
-	if err := c.client.RestartContainer(c.ID, timeoutSeconds); err != nil {
+	var exp []string
+	if err := c.Inspect(); err != nil {
```

Comment:
`Inspect` updates the internal container state, so if the container dies and then "restart" is called, we need a call to `Inspect` inbetween. I think Inspect() should probably just return something and not be cached.